src/uct/ib/ud/base/ud_ep.c

```diff
@@ -221,7 +225,7 @@ static void uct_ud_ep_purge_outstanding(uct_ud_ep_t *ep)
 
 static void uct_ud_ep_purge(uct_ud_ep_t *ep, ucs_status_t status)
 {
-    uct_ud_ep_tx_stop(ep);
+    uct_ud_ep_reset_max_psn(ep);
```

Comment:
it fixes doing `uct_iface_flush()`/`uct_ep_flush(LOCAL)` after `uct_ep_flush(CANCEL)` was done
before this fix, `uct_iface_flush()`/`uct_ep_flush(LOCAL)` returned `UCS_ERR_NO_RESOURCE` after canceling, since `no_window()` was true (it checks that `tx.psn == tx.max_psn`)

added a comment to clarify this fact